internal/controllers/nmc_reconciler.go

```diff
@@ -555,104 +565,132 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 	return errors.Join(errs...)
 }
 
-func (h *nmcReconcilerHelperImpl) UpdateNodeLabelsAndRecordEvents(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error {
-	node := v1.Node{}
-	if err := h.client.Get(ctx, types.NamespacedName{Name: nmc.Name}, &node); err != nil {
-		return fmt.Errorf("could not get node %s: %v", nmc.Name, err)
-	}
+type LabelPreparationHelper interface {
+	GetDeprecatedKernelModuleReadyLabels(deprecatedNodeModuleReadyLabels sets.Set[string], node v1.Node) sets.Set[string]
+	GetKernelModuleReadyLabels(nodeModuleReadyLabels sets.Set[types.NamespacedName], node v1.Node) sets.Set[types.NamespacedName]
+	GetSpecLabelsAndTheirConfigs(specLabels map[types.NamespacedName]kmmv1beta1.ModuleConfig, nmc *kmmv1beta1.NodeModulesConfig) map[types.NamespacedName]kmmv1beta1.ModuleConfig
+	GetStatusLabelsAndTheirConfigs(statusLabels map[types.NamespacedName]kmmv1beta1.ModuleConfig, nmc *kmmv1beta1.NodeModulesConfig) map[types.NamespacedName]kmmv1beta1.ModuleConfig
+}
+type labelPreparationHelperImpl struct {
+}
 
-	// get all the kernel module ready labels of the node
-	nodeModuleReadyLabels := sets.New[types.NamespacedName]()
-	deprecatedNodeModuleReadyLabels := sets.New[string]()
+func NewLabelPreparationHelper() LabelPreparationHelper {
```

Comment:
If we don't converge all those methods into `nmcReconcilerHelper` then let's make this function private (same as `newNmcReconcilerHelper`

Reply:
we use `NewLabelPreparationHelper` in cmd/manager/main.go line 124.
so I think that if we use this interface, we need this function to be Public